src/models/daily-registrations-report-data.ts

```diff
@@ -0,0 +1,91 @@
+import { ascending } from "d3-array";
+import { path as reportPath } from "./api-path";
+
+interface Result {
+  /**
+   * ISO8601 date-only string (YYYY-MM-DD)
+   */
+  date: string;
+  total_users: number;
+  fully_registered_users: number;
+}
+
+interface ProcessedResult {
+  date: Date;
+  totalUsers: number;
+  totalUsersCumulative: number;
+  fullyRegisteredUsers: number;
+  fullyRegisteredUsersCumulative: number;
+}
+
+enum DataType {
+  TOTAL_USERS,
+  TOTAL_USERS_CUMULATIVE,
+  FULLY_REGISTERED_USERS,
+  FULLY_REGISTERED_USERS_CUMULATIVE,
+}
+
+interface ProcessedRenderableData {
+  date: Date;
+  value: number;
+  type: DataType;
+}
+
+interface DailyRegistrationsReportData {
+  results: Result[];
+
+  /**
+   * ISO8601 string
+   */
+  finish: string;
```

Comment:
What do we use this for?

Reply:
We don't, but it's in the response JSON so I figured I'd document it